config/common.go

```diff
@@ -141,3 +142,14 @@ func (cs *commonStore) validate(cfg *model.Config) error {
 
 	return nil
 }
+
+// mergeConfig merges two configs together. The receiver's values are overwritten with the patch's
+// values except when the patch's values are nil.
+func (cs *commonStore) mergeConfig(patch *model.Config) (*model.Config, error) {
+	ret, err := utils.Merge(cs, patch)
+	if err != nil {
+		return nil, err
+	}
+	retC := ret.(model.Config)
```

Comment:
Seems fine to just `return &ret.(model.Config), nil` directly, if that's allowed.